examples/synthetics_example_test.go

```diff
@@ -143,125 +138,519 @@ func demonstrateSyntheticsTestsAPI() error {
+				Timeout:  10 * time.Second,
+				Count:    10,
+				Protocol: synthetics.PingProtocolICMP,
+			},
+			Traceroute: synthetics.TracerouteSettingsRequiredFields{
+				Timeout:  59999 * time.Millisecond,
+				Count:    5,
+				Delay:    100 * time.Millisecond,
+				Protocol: synthetics.TracerouteProtocolUDP,
+				Limit:    255,
+			},
+		},
+		Definition: synthetics.TestDefinitionFlowRequiredFields{
+			Type:          synthetics.FlowTestTypeCity,
+			Target:        "Warsaw",
+			Direction:     synthetics.DirectionSrc,
+			InetDirection: synthetics.DirectionDst,
+		},
+	})
+}
+
+func newMinimalURLTest(agentIDs []models.ID) *synthetics.Test {
+	return synthetics.NewURLTest(synthetics.URLTestRequiredFields{
+		BaseTestRequiredFields: synthetics.BaseTestRequiredFields{
+			Name:     "go-sdk-example-minimal-url-test",
+			AgentIDs: agentIDs,
+		},
+		Definition: synthetics.TestDefinitionURLRequiredFields{
+			Target: url.URL{
+				Scheme:   "https",
+				Host:     "www.example.com:443",
+				RawQuery: "dummy=query",
+			},
+			Timeout: time.Minute,
+		},
+	})
+}
+
+func newMinimalPageLoadTest(agentIDs []models.ID) *synthetics.Test {
+	return synthetics.NewPageLoadTest(synthetics.PageLoadTestRequiredFields{
+		BaseTestRequiredFields: synthetics.BaseTestRequiredFields{
+			Name:     "go-sdk-example-minimal-page-load-test",
+			AgentIDs: agentIDs,
+		},
+		Definition: synthetics.TestDefinitionPageLoadRequiredFields{
+			Target: url.URL{
+				Scheme:   "https",
+				Host:     "www.example.com:443",
+				RawQuery: "dummy=query",
+			},
+			Timeout: time.Minute,
+		},
+	})
+}
+
+func newMinimalDNSTest(agentIDs []models.ID) *synthetics.Test {
+	return synthetics.NewDNSTest(synthetics.DNSTestRequiredFields{
+		BaseTestRequiredFields: synthetics.BaseTestRequiredFields{
+			Name:     "go-sdk-example-minimal-page-load-test",
+			AgentIDs: agentIDs,
+		},
+		Definition: synthetics.TestDefinitionDNSRequiredFields{
+			Target:     "www.example.com",
+			Timeout:    time.Minute,
+			RecordType: synthetics.DNSRecordAAAA,
+			Servers:    []net.IP{net.ParseIP("192.0.2.213"), net.ParseIP("2001:db8:dead:beef:dead:beef:dead:beef")},
+			Port:       53,
+		},
+	})
+}
+
+func newMinimalDNSGridTest(agentIDs []models.ID) *synthetics.Test {
+	return synthetics.NewDNSGridTest(synthetics.DNSGridTestRequiredFields{
+		BaseTestRequiredFields: synthetics.BaseTestRequiredFields{
+			Name:     "go-sdk-example-minimal-page-load-test",
+			AgentIDs: agentIDs,
+		},
+		Definition: synthetics.TestDefinitionDNSGridRequiredFields{
+			Target:     "www.example.com",
+			Timeout:    time.Minute,
+			RecordType: synthetics.DNSRecordAAAA,
+			Servers:    []net.IP{net.ParseIP("192.0.2.213"), net.ParseIP("2001:db8:dead:beef:dead:beef:dead:beef")},
+			Port:       53,
+		},
+	})
+}
+
+func pickPrivateAgentID(agents []synthetics.Agent) (models.ID, error) {
+	var matchedIDs []models.ID
+	for _, a := range agents {
+		if a.Type == synthetics.AgentTypePrivate {
```

Comment:
Expecially for private agents, you need to also test that the `agent.status` is OK. If you want to be increase probability that the agent is actually operational, you can add test on the `last_authed` timestamp being less than 25 hours in the past (agents re-authorize every 24 hours).
Note that picking a non-operational agent causes test results to be unavailable, otherwise test creation (unfortunately) succeeds.

Reply:
Ok, but this example (`demonstrateSyntheticsAgentsAPI`) only shows available methods of Agents API. It's purpose is not finding working agents, display test results, etc.

The same for examples for synth tests (`TestDemonstrateSyntheticsTestsAPI`): they only do some API manipulation and immediately clean up after.